mcpwm_foc.c

```diff
@@ -3305,70 +3324,67 @@ static float correct_encoder(float obs_angle, float enc_angle, float speed, floa
 static float correct_hall(float angle, float speed, float dt) {
 	static int ang_hall_int_prev = -1;
 	float rpm_abs = fabsf(speed / ((2.0 * M_PI) / 60.0));
-	static bool using_hall = true;
 
 	// Hysteresis 5 % of total speed
 	float hyst = m_conf->foc_sl_erpm * 0.1;
-	if (using_hall) {
+	if (m_using_hall) {
 		if (rpm_abs > (m_conf->foc_sl_erpm + hyst)) {
-			using_hall = false;
+			m_using_hall = false;
 		}
 	} else {
 		if (rpm_abs < (m_conf->foc_sl_erpm - hyst)) {
-			using_hall = true;
+			m_using_hall = true;
 		}
 	}
 
-	if (using_hall) {
-		int ang_hall_int = m_conf->foc_hall_table[read_hall()];
-
-		// Only override the observer if the hall sensor value is valid.
-		if (ang_hall_int < 201) {
-			static float ang_hall = 0.0;
-			float ang_hall_now = (((float)ang_hall_int / 200.0) * 360.0) * M_PI / 180.0;
+	int ang_hall_int = m_conf->foc_hall_table[read_hall()];
 
-			if (ang_hall_int_prev < 0) {
-				// Previous angle not valid
-				ang_hall_int_prev = ang_hall_int;
```

Comment:
The hall logic is kept pretty much the same, the big heads up is that I deleted this line from the original code
`ang_hall_int_prev = ang_hall_int; 
`
Looks like it prevents the code to reach cases when ang_hall_int_prev is -1 or -2.